### DISCUSSION TITLE
Populating useFieldArray with context data

### DISCUSSION THREAD

**Discussion by HashiW** [react-hook-form/react-hook-form#2744] [Q&A] [answered]
First off React Hook Form is an excellent library and i'm really enjoying using it! I'm having some trouble populating the fieldArray with my React Context data like the example. I'm trying to combine a wizard and dynamic form and I want it to persist on refresh.  I'm following the Wizard/multistep funnel example but it wont persist.

I'm wondering how to use reset api or the default value to populate the field array with context data?

**Comment by bluebill1049** [comment]
do you have a simple codesandbox to illustrate your problem?

**Reply by HashiW** [reply]
My code is very similar to your dynamic form codesandbox example. In your video for dynamic forms you said it was possible to add default values. I was wondering how I could do so.

**Reply by HashiW** [reply]
The only significant difference between mine and the example is that I used the React Context API instead of little-state-machine.

**Reply by bluebill1049** [reply]
oh then it should be the same then, LSM use context under the hood.

**Reply by HashiW** [reply]
So how would I add a default value in your dynamic form example?

**Reply by bluebill1049** [reply]
can you provide a codesandbox of your work? you should be able to use `reset`API, or defaultValues at useForm

**Reply by HashiW** [reply]
My Issue is that I have a dynamic form like this example [https://codesandbox.io/s/cocky-night-mqe9g](url). Can you add default values to this example.  Adding default is significantly easier for static forms but I can't find a default value example for fieldArrays.

**Comment by HashiW** [comment, marked as answer]
The solution is to use the reset api on the useEffect. So when your application first renders the populate reset with your props like so. 
`
   useEffect(() => {
       reset(yourcontrol: props) 
    }, [])
`

Default value doesn't work when fetching data from API or resetting forms on the fly. In the future I hope this clarification is made more apparent on the Docs when addressing default Values . Thanks to @bluebill1049 for helping me with this.
